cloud-trail-local.tf

```diff
@@ -0,0 +1,212 @@
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": [
+        "kms:Decrypt",
+        "kms:ReEncryptFrom"
+      ],
+      "Resource": "*",
+      "Condition": {
+        "StringEquals": {
+          "kms:CallerAccount": "${data.aws_caller_identity.current.account_id}"
+        },
+        "StringLike": {
+          "kms:EncryptionContext:aws:cloudtrail:arn": "arn:aws:cloudtrail:*:${data.aws_caller_identity.current.account_id}:trail/*"
+        }
+      }
+    }
+  ]
+}
+POLICY
+}
+
+#create KMS alias for key created above
+resource "aws_kms_alias" "cloudtrail" {
+  name          = "alias/cloudtrail"
+  target_key_id = "${aws_kms_key.cloudtrail.key_id}"
+}
+
+#create the global cloudtrail
+resource "aws_cloudtrail" "global" {
+  name                       = "global"
+  s3_bucket_name             = "${aws_s3_bucket.global_cloudtrail.id}"
+  is_multi_region_trail      = true
+  enable_log_file_validation = true
+  kms_key_id                 = "${aws_kms_key.cloudtrail.arn}"
+}
+
+#create the bucket used by the above
+resource "aws_s3_bucket" "global_cloudtrail" {
+  bucket        = "${var.global_cloudtrail_bucket_name}"
+  force_destroy = false
+
+  lifecycle_rule {
+    id                                     = "logs-transition"
+    prefix                                 = ""
+    abort_incomplete_multipart_upload_days = 7
+    enabled                                = true
+
+    transition {
+      days          = 30
+      storage_class = "STANDARD_IA"
+    }
+
+    transition {
+      days          = 60
+      storage_class = "GLACIER"
+    }
+
+    expiration {
+      days = 365
+    }
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailAclCheck",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::${var.global_cloudtrail_bucket_name}"
+        },
+        {
+            "Sid": "AWSCloudTrailWrite",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::${var.global_cloudtrail_bucket_name}/*",
+            "Condition": {
+                "StringEquals": {
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+POLICY
+}
+
+
+# user module froom analytics-platform-ops
+module "aws_account_logging" {
+  source = "modules/aws_account_logging"
+
+  es_domain   = "${var.es_domain}"
+  es_port     = "${var.es_port}"
+  es_scheme   = "${var.es_scheme}"
+  es_username = "${var.es_username}"
+  es_password = "${var.es_password}"
```

Comment:
As we discussed on Slack. These sensitive variables should either be sourced from data_sources (from elastic) or held in SecretsManager.

Reply:
One for later I think.